### DISCUSSION TITLE
tutorials/scripting/c_sharp/c_sharp_differences

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#7] [User-contributed notes]
# tutorials/scripting/c_sharp/c_sharp_differences

This is a (incomplete) list of API differences between C# and GDScript. General differences: As explained in the C# basics, C# generally uses PascalCase instead of the snake_case used in GDScript a...

https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_differences.html

**Comment by 31** [comment]


**Reply by adamscott** [reply]
@31 Thanks for the feedback. I would suggest you to [create a bug report](https://github.com/godotengine/godot-docs/issues/new/choose) instead of commenting here for issues with the documentation, though, as these comments are intended as user notes as a complement to the documentation.

**Reply by 31** [reply]
> I would suggest you to [create a bug report](https://github.com/godotengine/godot-docs/issues/new/choose) instead of commenting here for issues with the documentation, though, as these comments are intended as user notes as a complement to the documentation.

Discussed at https://chat.godotengine.org/channel/documentation?msg=vnMEmF4JEnjMhRxJN. The rules seem to me to contradict that (as of writing), and hoping to get this clarified.

> ✅ What user-contributed notes are for
> Add a clarification or correct an invalid statement in the documentation, without having to open a pull request.
> (https://github.com/godotengine/godot-docs-user-notes/discussions/1 as of 2024-01-16)

**Reply by 31** [reply]
In any case, got around to filing https://github.com/godotengine/godot-docs/pull/8774 for this clarification. It's merged now, so deleted the start of the thread to clean up.

**Comment by EddieDover** [comment]
AABB IntersectsSegment returns a boolean in C# and a Vector3 in gdscript

**Reply by Calinou** [reply]
The reason this is the case is that the return type changed in GDScript between 3.x and 4.0, but this was missed in C#. Changing it now would break compatibility with existing projects that use the method in C#. See https://github.com/godotengine/godot/issues/103088 and https://github.com/godotengine/godot/pull/103152.
